modules/afamqp/compat/amqp-compat.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2019 Balabit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "syslog-ng.h"
+#include "compat/amqp-compat.h"
+
+#include <amqp_ssl_socket.h>
```

Comment:
The `lib/compat/compat.h` header has a nice description how to add additional compat layer, could you please follow that ? https://github.com/syslog-ng/syslog-ng/blob/5a28fa4d75b3b46d946c52cc74f66145e0437b12/lib/compat/compat.h#L44-L63

Reply:
@Kokan : I don't think we should follow that guide in case of modules...
That guide is for the `lib`. `lib` has several dependencies while modules are not.
I think `amqp-compat` is good enough.

Reply:
I agree with @lbudai on this one. I would like to keep this as simple as possible. I do not say a proper organization of code is useless, especially if the complexity increases over time. I just think this is good enough now, and it is better spending the APM elsewhere.

Reply:
@Kokan : you concrete requests would be to 
 * rename `compat/amqp-compat.h` to `compat/amqp.h` 
 * include `amqp.h` in `compat/amqp.h`
 * include only `compat/amqp.h` in `afqmqp.c` 
?

Reply:
The "include only compat/amqp.h in afamqp.", and also including `syslog-ng-config.h` instead of `syslog-ng.h`, as that could cause some issue, saidly I do not remember what exactly.

Reply:
syslog-ng.h was included because I needed glib.h due to gboolean. syslog-ng.h includes compat/glib.h too. syslog-ng-config.h would pull only the compiler switches.

Reply:
Because I cannot give proper reason, pleas go ahead with this PR.

Reply:
@Kokan : thx